### PR TITLE
feat(pagination): simple prop extend to object & add align prop

### PR DESCRIPTION
**Pull request by coding-ice**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/49499
https://github.com/ant-design/ant-design/issues/48203

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | simple prop extend to object & add align prop          |
| 🇨🇳 Chinese |simple属性，支持只读属性 & 增加 align属性 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -127,7 +127,7 @@
     "rc-menu": "~9.14.1",
     "rc-motion": "^2.9.2",
     "rc-notification": "~5.6.0",
-    "rc-pagination": "~4.0.4",
+    "rc-pagination": "~4.2.0",
```
Comment (iceweb1999) [inline package.json:130]: 4.1.0 有一个`align`属性还没有`merge`

Comment (coding-ice) [inline package.json:130]: 4.1.0 有一个align属性还没有merge

**Comment by yoyo837** [comment]
需要同时包含4.1.0和4.2.0的更新日志

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (afc163) [reply package.json:130]: 一直在这里做掉？

[the remark this answers is not in the document]
Comment (coding-ice) [reply package.json:130]: 没太听明白

[the remark this answers is not in the document]
Comment (afc163) [reply package.json:130]: align 属性的同步。

[the remark this answers is not in the document]
Comment (coding-ice) [reply package.json:130]: ok的，那我把align属性也同步到这个PR里面

File: components/pagination/Pagination.tsx [hunk 2]
```diff
@@ -115,6 +116,9 @@ const Pagination: React.FC<PaginationProps> = (props) => {
 
   const extendedClassName = classNames(
     {
+      [`${prefixCls}-start`]: align === 'start',
+      [`${prefixCls}-center`]: align === 'center',
+      [`${prefixCls}-end`]: align === 'end',
```
Comment (afc163) [inline components/pagination/Pagination.tsx:121]:
```suggestion
      [`${prefixCls}-${align}`]: true,
```

**Comment by coding-ice** [comment]
@afc163 大佬麻烦看下，又把`align`重新实现了一遍  
另外`List`和`Table`组件 `（pagination/align）` 这个PR中也需要一并修改吗

**Comment by afc163** [comment]
对，我建议一起改掉，可以简化原来的 align 实现。

**Comment by coding-ice** [comment]
`Table`组件的话这期`API`可能支撑不了，入参不太一样
image

File: components/list/index.tsx [hunk 3]
```diff
@@ -201,20 +201,15 @@ function List<T>({
...
+  const paginationContent = pagination && (
+    <div className={classNames(`${prefixCls}-pagination`)}>
       <Pagination
         {...paginationProps}
         onChange={onPaginationChange}
         onShowSizeChange={onPaginationShowSizeChange}
       />
     </div>
-  ) : null;
+  );
```
Comment (afc163) [inline components/list/index.tsx:212]: table 需要改么？

Comment (coding-ice) [reply components/list/index.tsx:212]: `table`的入参不太一样，这期`API`应该支撑不了

**Review by afc163 — APPROVED** [review]

**Pull request merged**